flink-table-store-common/src/main/java/org/apache/flink/table/store/lookup/hash/HashLookupStoreWriter.java

```diff
@@ -0,0 +1,486 @@
+            File tempIndexFile = indexFiles[keyLength];
+            DataInputStream tempIndexStream =
+                    new DataInputStream(
+                            new BufferedInputStream(new FileInputStream(tempIndexFile)));
+            try {
+                byte[] keyBuffer = new byte[keyLength];
+                byte[] slotBuffer = new byte[slotSize];
+                byte[] offsetBuffer = new byte[offsetLength];
+
+                // Read all keys
+                for (int i = 0; i < count; i++) {
+                    // Read key
+                    tempIndexStream.readFully(keyBuffer);
+
+                    // Read offset
+                    long offset = VarLengthIntUtils.decodeLong(tempIndexStream);
+
+                    // Hash
+                    long hash = MurmurHashUtils.hashBytesPositive(keyBuffer);
+
+                    boolean collision = false;
+                    for (int probe = 0; probe < count; probe++) {
+                        int slot = (int) ((hash + probe) % slots);
+                        byteBuffer.position(slot * slotSize);
+                        byteBuffer.get(slotBuffer);
+
+                        long found = VarLengthIntUtils.decodeLong(slotBuffer, keyLength);
+                        if (found == 0) {
+                            // The spot is empty use it
+                            byteBuffer.position(slot * slotSize);
+                            byteBuffer.put(keyBuffer);
+                            int pos = VarLengthIntUtils.encodeLong(offsetBuffer, offset);
+                            byteBuffer.put(offsetBuffer, 0, pos);
+                            break;
+                        } else {
+                            collision = true;
+                            // Check for duplicates
+                            if (Arrays.equals(keyBuffer, Arrays.copyOf(slotBuffer, keyLength))) {
+                                throw new RuntimeException(
+                                        String.format(
+                                                "A duplicate key has been found for for key bytes %s",
+                                                Arrays.toString(keyBuffer)));
+                            }
+                        }
+                    }
+
+                    if (collision) {
+                        collisions++;
+                    }
+                }
+
+                String msg =
+                        "  Max offset length: "
+                                + offsetLength
+                                + " bytes"
+                                + "\n  Slot size: "
+                                + slotSize
+                                + " bytes";
+
+                LOG.info("Built index file {}\n" + msg, indexFile.getName());
+            } finally {
+                // Close input
+                tempIndexStream.close();
+
+                // Close index and make sure resources are liberated
+                indexChannel.close();
```

Comment:
Does the `byteBuffer` need to clean?

Reply:
`FileChannel` is closed, there is no close in `byteBuffer`.